### DISCUSSION TITLE
fastAPI cannot take in both List[float] and Int or Float (cannot take both lists and single values) in the function GET/POST

### DISCUSSION THREAD

**Discussion by kiann00** [fastapi/fastapi#9174] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typing
from typing import List
from fastapi import Query
from fastapi import FastAPI

@appOne.get("/sabr/calib_test2")
    def calib_test2(x1: List[float] = [1, 2, 3], x2: List[float] = [4, 5, 6],  type_: List[int] = [1]):
        s1 = np.array(x1)
        s2 = np.array(x2)
        corr = np.corrcoef(x1, x2)[0][1]
        type_ = type_[0]
        if type_ == 1:
            return {"x1": x1, "x1_sum": s1.sum(), "x2": x2, "x2_sum": s2.sum(), "size_x1": s1.shape, "corr": corr,\
            "type_": type_}
        elif type_ == 2:
            return ['test', x1, s1.sum(), x2, s2.sum(), s1.shape, corr, type_]
        else:
            return [0, 0, 0]
```

### Description

Using the example code, it seems one cannot have the fastAPI to read both list, and single (float or otherwise) input data into the function.

I use the below in the jupyter notebook :

```
url_ = "http://127.0.0.1:8000/sabr/"
func_ = "calib_test2"
item1 = [1, 2, 3, 4, 5]
item2 = [4, 5, 6, 7, 8]
params = {"x1": [1, 2, 3, 4], "x2": [1, 2, 3, 4], "type_": 1}
print('\n', requests.get(url_ + func_, json=params).json())
params = {"x1": [1, 2, 3, 4], "x2": [1, 2, 3, 4], "type_": 2}
print(requests.get(url_ + func_, json=params).json())
```

I am getting the error message below : 

 {'detail': [{'loc': ['body', 'type_'], 'msg': 'value is not a valid list', 'type': 'type_error.list'}]}
{'detail': [{'loc': ['body', 'type_'], 'msg': 'value is not a valid list', 'type': 'type_error.list'}]}

### Operating System

Windows

### FastAPI Version

0.92

### Python Version

3.9

**Comment by ebottos94** [comment]
Hi @kiann00, first of all if you want to set this validation, you must use post method. Second, if you want to use multiple validation type in one field you can use Union (or | operator from python 3.10+).

For example : 

```python
from fastapi import FastAPI
from typing import List, Union

app = FastAPI()

@app.post("/sabr/calib_test2")
def calib_test2(type_: Union[List[int], int] = [1]):
    return {"type_" : type_}
```
